### ISSUE TITLE
Change all `prefetchQuery` calls to `ensureQueryData` or `fetchQuery`

### ISSUE THREAD

**Issue by david-crespo** [oxidecomputer/console#2616]
`prefetchQuery` eats errors, which has the very annoying consequence that any error we run into in a loader manifests as the `ensurePrefetched` invariant failure in the component. We never see the real error, only the invariant failure. This means opening the browser console tells you nothing about what went wrong. See #2614 for an example of something made much harder to debug by this. It might also help a little with #2573.

### Options

* `fetchQuery` ([docs](https://tanstack.com/query/v5/docs/reference/QueryClient/#queryclientfetchquery), [source](https://github.com/TanStack/query/blob/1eebfc9d3355c411b16146cc8df76c18e63d59ad/packages/query-core/src/queryClient.ts#L398-L427)) fetches data unless it's cached and non-stale.

* `prefetchQuery` ([docs](https://tanstack.com/query/v5/docs/reference/QueryClient/#queryclientprefetchquery), [source](https://github.com/TanStack/query/blob/1eebfc9d3355c411b16146cc8df76c18e63d59ad/packages/query-core/src/queryClient.ts#L428C1-L438C4) is the same as `fetchQuery` except it eats errors and returns nothing. 😶 

  ```ts
  prefetchQuery<...>(
    options: FetchQueryOptions<TQueryFnData, TError, TData, TQueryKey>,
  ): Promise<void> {
    return this.fetchQuery(options).then(noop).catch(noop)
  }
  ```
  
* `ensureQueryData` ([docs](https://tanstack.com/query/v5/docs/reference/QueryClient/#queryclientensurequerydata), [source](https://github.com/TanStack/query/blob/1eebfc9d3355c411b16146cc8df76c18e63d59ad/packages/query-core/src/queryClient.ts#L140-L164)) only calls `fetchQuery` if there is no data in the cache, regardless of staleness. If `revalidateIfStale` is `true` (default `false`), it will call `prefetchQuery` to revalidate if there is data in the cache but it's stale. By default, it will not do this, meaning it will simply return cached data if it's present, regardless of staleness.

I'm a little torn between `fetchQuery` and `ensureQueryData`. I'm not sure we need the extra logic in `ensureQueryData` — I think when we run a loader, we do want to make sure the data is fresh because it's a page navigation. I think we want the app to behave as much like a traditional web app as possible. I also don't want to have to pass in `revalidateIfStale: true` everywhere, so I'd probably add a `loaderPrefetch` function to the `QueryClient` object if I wanted to use `ensureQueryData`.

**Comment by askfongjojo** [comment]
Not sure if this error I see on rack2 is related to this ticket, or it should be filed as a separate issue:

<img width="1208" height="769" alt="Image" src="https://github.com/user-attachments/assets/b729bdf1-f820-409a-a3a1-5dbd1b5cdb56" />

**Comment by david-crespo** [comment]
Nope, different issue. Looking into it now. I'll write an issue if I can't fix it.

Update: rack2 is accidentally running a very old console, so it's running into API compatibility issues.

**Comment by askfongjojo** [comment]
Yes, the old console code causes the instance list to fail but what I pasted above is the anti-affinity group list: https://oxide.sys.rack2.eng.oxide.computer/projects/product-assurance/affinity

**Comment by david-crespo** [comment]
Ah. I assumed it was the same issue. Fixed in #3279.
